src/Poisson2DSparseMatrix.cpp

```diff
@@ -139,17 +145,75 @@ namespace blitzdg{
                                                                  // for '+' traces.
 
                 real_matrix_type gDxM(NGauss, Np), gDyM(NGauss, Np),
-                    gDxP(NGauss, Np), gDyP(NGauss, Np);
+                    gDxP(NGauss, Np), gDyP(NGauss, Np), gDr(NGauss, Np), gDs(NGauss, Np);
 
-                // Evaluate spatial derivatives of  Lagrange basis function at Gauss nodes
-                dg.computeDifferentiationMatrices(xLocal, yLocal, gVM, gDxM, gDyM);
-                dg.computeDifferentiationMatrices(xLocal2, yLocal2, gVP, gDxP, gDyP);
+                // Compute x and y differentiation matrices that evaluate
+                // derivatives at Gauss nodes.
+                const real_matrix_type& gInterp = gCtx.Interp();
+
+                gDr = blitz::sum(gInterp(ii, kk)*Dr(kk, jj), kk);
+                gDs = blitz::sum(gInterp(ii, kk)*Ds(kk, jj), kk);
+
+                real_vector_type xr1(NGauss), xs1(NGauss), yr1(NGauss), ys1(NGauss),
+                    J1(NGauss), rx1(NGauss), sx1(NGauss), ry1(NGauss), sy1(NGauss);
+
+                // Compute geometric factors on this element.
+                xr1 = blitz::sum(gDr(ii, jj)*xLocal(jj), jj);
+                xs1 = blitz::sum(gDs(ii, jj)*xLocal(jj), jj);
+
+                yr1 = blitz::sum(gDr(ii, jj)*yLocal(jj), jj);
+                ys1 = blitz::sum(gDs(ii, jj)*yLocal(jj), jj);
+
+                J1 = -xs1*yr1 + xr1*ys1;
+
```

Comment:
Any chance the `J1` can be zero here? Is that something we should check and issue an error if so?

Reply:
If J1 is zero or changes sign it means it's somehow non-uniquely mappable to the standard element, which is bad. We probably do want to issue an error if any element is zero to within some tolerance.

Reply:
This was apparently resolved in a separate commit sometime...